### PR TITLE
feat(devops): Add azureAutoComplete config option

### PR DESCRIPTION
**Pull request by hillerstorm**
Adds `azureAutoComplete` as a configuration option to enable setting Azure DevOps PRs to auto-complete.
This includes a minor refactor to `createPr` which adds platform specific options as the last argument (`statusCheckVerify` and `azureAutoComplete` for now).

### CODE REVIEW

File: lib/workers/repository/onboarding/pr/index.js [hunk 1]
```diff
@@ -115,12 +115,17 @@ If you need any further assistance then you can also [request help here](${
     if (config.dryRun) {
       logger.info('DRY-RUN: Would create onboarding PR');
     } else {
+      const platformOptions = {
```
Comment (rarkins) [inline lib/workers/repository/onboarding/pr/index.js]: it's not necessary to set it for onboarding. We can even just leave it undefined

**Comment by JamieMagee** [comment]
@hillerstorm Do you think it would be possible to reuse the existing `automerge` configuration option instead of creating a new platform specific option? This is a genuine question, I don't know if it's possible or required, but it might allow us to use `automergetype` as well

**Comment by hillerstorm** [comment]
> @hillerstorm Do you think it would be possible to reuse the existing `automerge` configuration option instead of creating a new platform specific option? This is a genuine question, I don't know if it's possible or required, but it might allow us to use `automergetype` as well

We discussed that a bit, the difference with auto complete on Azure PRs is that it's only an `auto merge` if there's no branch policies set up, otherwise it's just a way to specify *how* you want your changes merged when it's okay to do so. A PR set to auto-complete can be cancelled by the reviewer for example, in which case it has no effect and is treated as a manual PR.

This way was also much less work required to implement :) We could switch to using `automergeType`, but in the future this might get more flags related to it (do you want to squash commits, do you want a specific merge commit message etc).
There is also a TODO set up in workers/branch/index.js which states that `ensurePr` should check for automerge, which I guess would have to be tackled if we added azure-auto-complete as an `automergeType`, because of how the checks are done today, assuming a lot when checking for `automerge` without even considering the `automergeType` in some cases.

**Comment by hillerstorm** [comment]
@rarkins anything more to do on this one? :)

**Comment by rarkins** [comment]
LGTM

**Comment by rarkins** [comment]
We should probably work out the best semantic commit scope for Azure DevOps - `azure`, `azure-devops`, or `devops` ? I think the latter as used here is too confusing for anyone who may be a non-Azure DevOps user and doesn't even *know* the term "devops" might refer to Azure. So that leaves "azure" or "azure-devops"

**Pull request merged**